### PR TITLE
Faults implementation

### PR DESCRIPTION
**Pull request by Christian-Fisher**
Added both hard and transient faults to moving and doors opening/closing respectively. Currently if the elevator enters a fault state, it will simply print it and then go on with its task. This WILL result in incorrect behaviour until the hard and transient states are implemented

### CODE REVIEW

File: src/Elevator.java [hunk 1]
```diff
@@ -204,25 +246,37 @@ public void run() {
...
-
+		int startFloor = this.getCurrentFloor(); // Saves the position of the elevator before attempting a move.
 		this.updateCurrentFloor();
-		System.out.println("Elevator "+elID+": At floor: " + getCurrentFloor());
-		Thread.sleep(2000);
+		Thread.sleep(3000);
+		if (this.getCurrentFloor() == startFloor) { // If the elevator has not moved after 3 seconds, it is stuck and
+													// therefore is hard faulting
+			System.out.println("Elevator " + this.getElID()+ "in hard fault state");
+//			currState = HARDFAULTSTATE;
```
Comment (Christian-Fisher) [inline src/Elevator.java:255]: This line will need to be changed and uncommented when the hard fault state is implemented

File: src/Elevator.java [hunk 2]
```diff
@@ -170,25 +188,49 @@ public void run() {
...
+
 				}
-	
+
 				case ReqFloorDoorsOpened: {
+					try {
+						this.setDoors(false);
+						Thread.sleep(1100);
+						if (this.isDoorsOpen()) {
+//							currState=TRANSIENTSTATE;
```
Comment (Christian-Fisher) [inline src/Elevator.java:218]: This line will need to be changed and uncommented when the transient fault state is implemented

File: src/Elevator.java [hunk 3]
```diff
@@ -170,25 +188,49 @@ public void run() {
...
 					break;
 				}
-	
+
 				case ArriveReqFloor: {
+					try {
+						this.setDoors(true);
+						Thread.sleep(1100);
+						if (!this.isDoorsOpen()) {
+//							currState=TRANSIENTSTATE;
```
Comment (Christian-Fisher) [inline src/Elevator.java:200]: This line will need to be changed and uncommented when the transient fault state is implemented

**Comment by dhyanpathak** [comment]
Looks good to me!

**Review by karanvirchaudhary — COMMENTED** [review]
Looks good to me

**Pull request merged**